common/djangoapps/course_modes/views.py

```diff
@@ -253,17 +253,28 @@ def get(self, request, course_id, error=None):  # lint-amnesty, pylint: disable=
         if deadline:
             formatted_audit_access_date = strftime_localized_html(deadline, 'SHORT_DATE')
             context['audit_access_deadline'] = formatted_audit_access_date
-        fbe_is_on = deadline and gated_content
+        full_fbe_is_on = deadline and gated_content
+        partial_fbe_is_on = deadline or gated_content
 
+        # Route to correct Track Selection page.
         # REV-2133 TODO Value Prop: remove waffle flag after testing is completed
         # and happy path version is ready to be rolled out to all users.
         if VALUE_PROP_TRACK_SELECTION_FLAG.is_enabled():
-            # First iteration of happy path does not handle errors. If there are enrollment errors for a learner that is
-            # technically considered happy path, old Track Selection page will be displayed.
-            if not error:
-                # Happy path conditions.
-                if verified_mode and fbe_is_on and not enterprise_customer:
-                    return render_to_response("course_modes/track_selection.html", context)
+            if not error:  # TODO: Remove by executing REV-2355
+                if not enterprise_customer:  # TODO: Remove by executing REV-2342
+                    if full_fbe_is_on:
+                        return render_to_response("course_modes/track_selection_types/full_fbe.html", context)
+                    elif partial_fbe_is_on:
+                        return render_to_response("course_modes/track_selection_types/partial_fbe.html", context)
+                    else:
+                        # If the course has started redirect to course home instead
+                        if course.has_started():
+                            return redirect(reverse(
+                                'openedx.course_experience.course_home',
+                                kwargs={'course_id': course_key}))
+                        return redirect(reverse('dashboard'))
+
+        # If error or enterprise_customer, failover to old choose.html page
```

Comment:
I'm a little confused about this block: this sort of redirect is included is this is a post request, for example line 320: 
https://github.com/edx/edx-platform/blob/6ed1ae05cbc972620f42e67d9d0642d8ae6e7b0f/common/djangoapps/course_modes/views.py#L320

This block is included if it's a get request, but only if it also doesn't match any of the conditions above? Could you say a little more about why it's being added here also?

Reply:
Yes, of course! This is basically simulating a 302 redirect, doing the work case 5 is supposed to do ("If the course does not have a verified mode, [5.] do not show partial_fbe.html or full_fbe.html. Instead, redirect to course page if the user has access to the course, or the dashboard if they do not yet have access to the course.")

Reply:
ah I gotcha! I got confused because of the comment "If the course has started redirect to course home instead"- it sounds like instead we want some condition/comment that this will only happen in the case where there's no verified mode available. Also I think we'll know that up front so it could come before the others and potentially save some time.